### ISSUE TITLE
Error connecting to Subsurface cloud storage

### ISSUE THREAD

**Issue by marcvs** [subsurface/subsurface#3946]
### Describe the issue:
- [x ] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:
- I've created the cloud account
- I can log in to https://ssrf-cloud-eu.subsurface-divelog.org/user/<email-address>/
- I can `git clone clone https://ssrf-cloud-us.subsurface-divelog.org/git/<email-address`; Result being:
```
Username for 'https://ssrf-cloud-us.subsurface-divelog.org': <email-address>
Password for 'https://marcus@hardt-it.de@ssrf-cloud-us.subsurface-divelog.org': 
remote: Enumerating objects: 275, done.
remote: Counting objects: 100% (275/275), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 275 (delta 149), reused 273 (delta 148)
Receiving objects: 100% (275/275), 124.51 KiB | 311.00 KiB/s, done.
Resolving deltas: 100% (149/149), done.
warning: remote HEAD refers to nonexistent ref, unable to checkout
```
- The resulting git-repo is empty
- I can see the dives (synced from phone) in "https://ssrf-cloud-eu.subsurface-divelog.org/user/<email-address>"
- yet, subsurface complains (on "file -> open cloud storage"), showing this error on the cmdline:
```
INFO: git storage: clone of https://ssrf-cloud-us.subsurface-divelog.org/git/<email-address>
ERROR: Error connecting to Subsurface cloud storage
```

### Operating system:
- OS: `Linux - debian/testing - en_US.UTF-8`
- Device: `Lenovo Thinkpad`
- Other device: `works on my mobile`

### Subsurface version:
- Version: `5.0.10`
- No other version ever tried
- Using debian packages from `https://dfx.at/subsurface-debian sid/main amd64 Packages`

### Steps to reproduce:
1. start subsurface (from cmdline)
2. File -> Preferences -> Cloud: Specify email address and password (verified **unencrypted**  entries in config file
3. File -> Open Cloud Storage
4. Find error on console

### Current behavior:
I get the error

### Expected behavior:
Load / display my (about 10) dives

### Additional information:
No applicable

**Comment by dirkhh** [comment]
It's hard to help you debug this without knowing the email address that you used to create the cloud storage account.
The most likely scenario is an incorrect password on your debian machine.
When you enter the preferences and go to the cloud tab, does it tell you "Subsurface cloud storage (credentials verified)" at the top?

**Comment by marcvs** [comment]
Yep it says "Subsurface cloud storage (credentials verified)"
The email address is `<first name>@<last name>-it.de`
And the name is Marcus Hardt

**Comment by dirkhh** [comment]
Yeah, I can see quite a few errors in the log that indicate that you are trying to use the Subsurface cloud as a git backend that you access with command line tools. That is not what this infrastructure is designed for (and there are a few guard rails to make this harder). The backend is exclusively designed to be used with the Subsurface apps.
The errors that I can see all appear to be coming from command line access attempts to the cloud. And so far I haven't been able to spot access attempts from Subsurface on Debian that actually fail in all that noise.

I would like to politely request that you stop trying to use the servers in ways they aren't intended to be used.

I'm happy to help you get Subsurface and Subsurface-mobile working with the backend.

On Debian, run Subsurface from the command line and redirect stdout/stderr to a file. Try to connect to the cloud (make sure that the you try to take the cloud connection online - there needs to be a checkmark by that menu entry). Then exit Subsurface. Please send me the resulting log file (unedited) via email to dirk at hohndel dot org

**Comment by AlexTR85** [comment]
I have the same problem.
![image](https://github.com/subsurface/subsurface/assets/56876938/e21ccf80-8159-432a-8476-97bead58c1ab)
 
For me is Windows 10, and latest version subsurface.

**Comment by marcvs** [comment]
I've only tried using git, after subsurface errors suggested this could be 
a way to find out why things failed. I promise that subsurface was used for 
access most of the time.

I've also only used git clone twice, a few pulls, and no other command. 

The stderr/stdout was already what i had sent in the initial report (dont have laptop with subsurface with me)

**Comment by marcvs** [comment]
This is the stdout/stderr on the console:
```
cmarcus@nemo:~$ subsurface
QObject::connect(QQuickWindow, QDeclarativeGeoMap_QML_6): invalid nullptr parameter
failed to connect to https://ssrf-cloud-eu.subsurface-divelog.org/ next server to try:  https://ssrf-cloud-us.subsurface-divelog.org/
INFO: git storage: clone of https://ssrf-cloud-us.subsurface-divelog.org/git/marcus@hardt-it.de failed
ERROR: Error connecting to Subsurface cloud storage
```

**Comment by nord-** [comment]
There is something fishy with the cloud storage. Last weekend I couldn't open it at all from the mobile app, but when I got home I could connect with the desktop version (Windows), and uploaded with the two dives from my dive computer.
Earlier this week I could sync the mobile app again but noticed that the two latest dives were missing and today I tried to sync from the desktop app to the cloud again and got this:
![image](https://github.com/subsurface/subsurface/assets/3777600/61b48b7b-1f4a-4dd7-a3b5-2c5a03ff17e9)
and
![image](https://github.com/subsurface/subsurface/assets/3777600/8300118f-747d-47d9-93bc-a2a8cdfd2dd9)

I can see my dives at https://ssrf-cloud-eu.subsurface-divelog.org/user/rickard@nord.cc/dives.html but the two from last weekend are missing there as well, so it does not seem to sync.

Now, after 10-15 mins this appeared (I have "No proxy" configured in my settings):
![image](https://github.com/subsurface/subsurface/assets/3777600/a2556b87-49b0-46e3-a0fc-48004b0c117e)
